server/src/main/java/org/elasticsearch/index/mapper/vectors/SparseVectorFieldMapper.java

```diff
@@ -177,15 +177,11 @@ public void parse(DocumentParserContext context) throws IOException {
             for (Token token = context.parser().nextToken(); token != Token.END_OBJECT; token = context.parser().nextToken()) {
                 if (token == Token.FIELD_NAME) {
                     feature = context.parser().currentName();
-                    if (feature.contains(".")) {
-                        throw new IllegalArgumentException(
-                            "[sparse_vector] fields do not support dots in feature names but found [" + feature + "]"
-                        );
-                    }
                 } else if (token == Token.VALUE_NULL) {
                     // ignore feature, this is consistent with numeric fields
                 } else if (token == Token.VALUE_NUMBER || token == Token.VALUE_STRING) {
-                    final String key = fullPath() + "." + feature;
+                    // Use a delimiter that won't collide with subfields
+                    final String key = fullPath() + ".." + feature;
```

Comment:
Oooo, sneaky

Reply:
If we allow dots in features it would be safer to escape the original form:
```suggestion
                    // escape the dots in the feature name
                    final String key = fullPath() + "." + feature.replace(".", "\\.");
```